identity-iota/src/tangle/client.rs

```diff
@@ -183,6 +198,19 @@ impl Client {
       .await
       .map_err(Into::into)
   }
+
+  /// Adds message version flag and encoding flag at the beginning of arbitrary message data.
+  fn add_flags_to_message(
+    mut message_data: Vec<u8>,
+    message_version: DIDMessageVersion,
+    encoding: DIDMessageEncoding,
+  ) -> Vec<u8> {
+    let message_version_flag = message_version as u8;
+    let encoding_flag = encoding as u8;
+
+    message_data.splice(0..0, [message_version_flag, encoding_flag].iter().cloned());
```

Comment:
```suggestion
message_data.splice(0..0, [message_version_flag, encoding_flag].into_iter());
``` 
This works now because we are using the 2021 edition of Rust.

Reply:
Just the following worked due to the bounds on splice (and Rust 2021):
```rust
message_data.splice(0..0, [message_version_flag, encoding_flag]);
```